ansible/devutil/devices/ansible_hosts.py

```diff
@@ -541,14 +545,23 @@ def _check_results(self, caller_info, module_info, results, module_ignore_errors
 
         if isinstance(self, AnsibleHosts):
             if isinstance(module_info, dict):
+                unreachable = not all([res["reachable"] for res in results.values()])
                 failed = any([res["failed"] for res in results.values()])
             else:
+                unreachable = not all(
+                    [all([res["reachable"] for res in module_results]) for module_results in results.values()]
+                )
                 failed = any([any([res["failed"] for res in module_results]) for module_results in results.values()])
         elif isinstance(self, AnsibleHost):
             if isinstance(module_info, dict):
+                unreachable = results["unreachable"]
                 failed = results["failed"]
             else:
+                unreachable = not any([res["reachable"] for res in results])
                 failed = any([res["failed"] for res in results])
+
+        if unreachable:
```

Comment:
Where defined this var, if the 'if' and 'elif' statement not reached, what does this var equal to? If is None, suggest to define it explicitly.

Reply:
Variables `unreachable` and `failed` are defined in `if...else...` caluses between line 542 and 563. But yes, it is a good idea to explicitly initialize them. Just pushed a new commit. This change also added another check of the current instance type.